### PR TITLE
Add notes for n18242

### PR DESCRIPTION
**Pull request by jnewbery**
Initial notes provided by @ariard. I've made some edits and modifications.

### CODE REVIEW

File: _posts/2020-06-17-#18242.md [hunk 1]
```diff
@@ -10,14 +10,80 @@ status: upcoming
...
 ## Questions
 
---->
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK?](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)
+   (Don't forget to put your PR review on GitHub.)
+
+2. What's your process to review a PR implementing a new BIP? Did you read the
+   BIP or the code first? How can you ensure that BIP is correct? How can you ensure
+   that code implementats the BIP a correctly?
```
Comment (ariard) [inline _posts/2020-06-17-#18242.md:74]: *implements

Comment (jnewbery) [reply _posts/2020-06-17-#18242.md:74]: thanks. Fixed this and a few other typos

**Comment by jnewbery** [comment]
> question on DoS vector, performance has been dropped

I think there's already enough to discuss just on the PR, without adding questions about potential further improvements :)

File: _posts/2020-06-17-#18242.md [hunk 2]
```diff
@@ -10,14 +10,80 @@ status: upcoming
...
+**Note:** in this PR Review club we will review the first three commits only
+(`Expose MAC length in chacha_poly_ahead_h`, `Add BIP324 short-IDs to
+protocol.h/.cpp` and `Add BIP324 v2 transport serializer and deserializer`).
 
-<!-- TODO: Add notes and questions
 ## Notes
 
+- Today's PR is part of the implementation of [BIP324: Version 2 Peer-to-Peer
+  Message Transport
+  Protocol](https://gist.github.com/jonasschnelli/c530ea8421b8d0e80c51486325587c52A),
```
Comment (amitiuttarwar) [inline _posts/2020-06-17-#18242.md:21]: I'm getting invalid link

File: _posts/2020-06-17-#18242.md [hunk 3]
```diff
@@ -10,14 +10,80 @@ status: upcoming
...
+  network. The previous proposal ([BIP
+  151](https://github.com/bitcoin/bips/blob/master/bip-0151.mediawiki)) was
+  also authored by Jonas Schnelli and has been withdrawn.
+
+- In BIP 324, messages are encrypted by the stream cipher
+  [ChaCha20](https://tools.ietf.org/html/rfc7539) with a Poly1305 [Message
+  Authentication
+  Code](https://en.wikipedia.org/wiki/Message_authentication_code).
+
+- A BIP 324 session starts with an [Eliptic Curve Diffie Hellman key
```
Comment (amitiuttarwar) [inline _posts/2020-06-17-#18242.md:41]: typo: Elliptic

File: _posts/2020-06-17-#18242.md [hunk 4]
```diff
@@ -10,14 +10,80 @@ status: upcoming
...
+  - [#15206: Immediately disconnect on invalid net message checksum](https://github.com/bitcoin/bitcoin/pull/15206) (discussed in a [previous Bitcoin Core PR Review Club](/15206))
+  - [#15519: Add Poly1305 implementation](https://github.com/bitcoin/bitcoin/pull/15519)
+  - [#15649: Add ChaCha20Poly1305@Bitcoin AEAD](https://github.com/bitcoin/bitcoin/pull/15649)
+  - [#16202: Refactor network message deserialization](https://github.com/bitcoin/bitcoin/pull/16202)
+  - [#16562: Refactor message transport packaging](https://github.com/bitcoin/bitcoin/pull/16562)
+
+- [Jonas Schnelli's talk](https://github.com/kanzure/diyhpluswiki/blob/master/transcripts/breaking-bitcoin/2019/p2p-encryption.mdwn)
+  at Breaking Bitcoin 2019 has additional information on BIP 324.
+
+- A further step to prevent eavesprodding on the P2P network would be to add
```
Comment (amitiuttarwar) [inline _posts/2020-06-17-#18242.md:61]: eavesdropping :)

File: _posts/2020-06-17-#18242.md (hunk 3 above)
Comment (jnewbery) [reply _posts/2020-06-17-#18242.md]: fixed!

File: _posts/2020-06-17-#18242.md (hunk 4 above)
Comment (jnewbery) [reply _posts/2020-06-17-#18242.md]: fixed!

File: _posts/2020-06-17-#18242.md (hunk 2 above)
Comment (jnewbery) [reply _posts/2020-06-17-#18242.md]: fixed

**Pull request merged**